src/js/services/AuthTokenStorage.js

```diff
@@ -0,0 +1,52 @@
+class AuthTokenStorage {
+
+  constructor() {
+    this._createWorkerInstance();
+  }
+
+  _createWorkerInstance() {
+    if(window && window.SharedWorker) {
+        this.worker = new window.SharedWorker('/js/tripleo_ui_token_worker.js');
+        this.worker.port.start();
+    }
+  }
+
+  /**
+   * Get the currently stored token ID.
+   * @returns {String|Boolean} The token Id or false.
+   */
+  getTokenId(cb) {
+    let tokenId = sessionStorage.getItem('keystoneAuthTokenId');
+
+    // If there was a tokenId stored in sessionStorage,
+    // just call the cb and leave.
+    if(tokenId) {
+        return cb(tokenId);
+    }
+
+    // Create callback for temporary worker message listener.
+    let that = this;
```

Comment:
This should probably not be necessary according to this https://github.com/airbnb/javascript#22.5